AmplifyPlugins/Auth/AWSAuthPlugin/Operations/AWSAuthSignOutOperation.swift

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+public class AWSAuthSignOutOperation: AmplifyOperation<AuthSignOutRequest,
+    Void,
+    Void,
+    AmplifyAuthError>,
+AuthSignOutOperation {
+    
+    let authenticationProvider: AuthenticationProviderBehavior
+    
+    init(_ request: AuthSignOutRequest,
+         authenticationProvider: AuthenticationProviderBehavior,
+         listener: EventListener?) {
+        
+        self.authenticationProvider = authenticationProvider
+        super.init(categoryType: .auth,
+                   eventName: HubPayload.EventName.Auth.webUISignIn,
+                   request: request,
+                   listener: listener)
+    }
+    
+    override public func main() {
+        if isCancelled {
+            finish()
+            return
+        }
+
+        authenticationProvider.signOut(request: request) { [weak self]  result in
+            
+            guard let self = self else { return }
```

Comment:
Yes, one way I can think of is the operation getting cancelled and removed out from the queue? Making self weak is more of a precaution to avoid any memory retention or cycles.